openshift/e2e_oncluster_tests.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Runs basic lifecycle E2E tests against kn func cli for a given language/runtime.
+# By default it will run e2e tests against 'func' binary, but you can change it to use 'kn func' instead
+#
+# The following environment variable can be set in order to customize e2e execution:
+#
+# E2E_USE_KN_FUNC    When set to "true" indicates e2e to issue func command using kn cli.
+#
+# E2E_REGISTRY_URL   Indicates a specific registry (i.e: "quay.io/user") should be used. Make sure
+#                    to authenticate to the registry (i.e: docker login ...) prior to execute the script
+#                    By default it uses "ttl.sh" registry
+#
+# E2E_FUNC_BIN_PATH  Path to func binary. Derived by this script when not set
+#
+# E2E_RUNTIMES       List of runtimes (space separated) to execute TestRuntime.
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+#source "$(go run knative.dev/hack/cmd/script e2e-tests.sh)"
+
+pushd "$(dirname "$0")/.."
+
+export E2E_REGISTRY_URL="${E2E_REGISTRY_URL:-ttl.sh/knfuncci$(head -c 128 </dev/urandom | LC_CTYPE=C tr -dc 'a-z0-9' | fold -w 6 | head -n 1)}"
```

Comment:
Yea.. I tried internal OCP registry either without success. I don't recall the error message, but it was sort of attempting to use the internal registry of the cluster running the test.., instead of the remote cluster (owned by serverless-ci) claimed for the test.. I can retry it again... 

and yea.. `ttl.sh` here is just the default. We can pass it to use the internal registry on the other [PR](https://github.com/openshift/release/pull/51325)

Reply:
Wouldn't `head -c 128 </dev/urandom | LC_CTYPE=C tr -dc 'a-z0-9' | head -c 6` work too?

Reply:
that fold looks superfluous

Reply:
yea.. true.. it can be simplified. I'll update on another PR